### PR TITLE
#194: SubText

### PR DESCRIPTION
**Pull request by fabriciofx**
As per #194:

`SubText` introduced.

### CODE REVIEW

**Comment by 0crat** [comment]
@fabriciofx I'm not managing this repo, remove the [webhook](https://github.com/yegor256/cactoos/settings/hooks) or contact me in [Slack](http://www.zerocracy.com) //cc @yegor256

File: src/main/java/org/cactoos/text/SubText.java [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+    @Override
+    @SuppressWarnings("PMD.AvoidCatchingGenericException")
+    public String asString() throws IOException {
+        try {
+            return this.origin.asString().substring(
+                this.start.asValue(),
+                this.end.asValue()
+            );
+            // @checkstyle IllegalCatchCheck (1 line)
+        } catch (final Exception ex) {
```
Comment (yegor256) [inline src/main/java/org/cactoos/text/SubText.java]: @fabriciofx use `UncheckedText` instead

File: src/test/java/org/cactoos/text/SubTextTest.java [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+import org.junit.Test;
+
+/**
+ * Test case for {@link SubText}.
+ * @author Fabricio Cabral (fabriciofx@gmail.com)
+ * @version $Id$
+ * @since 0.9
+ * @checkstyle JavadocMethodCheck (500 lines)
+ */
+@SuppressWarnings("PMD.AvoidDuplicateLiterals")
```
Comment (yegor256) [inline src/test/java/org/cactoos/text/SubTextTest.java]: @fabriciofx why test methods are using the same string literals?

**Comment by yegor256** [comment]
@fabriciofx see above

File: src/test/java/org/cactoos/text/SubTextTest.java (hunk 2 above)
Comment (fabriciofx) [inline src/test/java/org/cactoos/text/SubTextTest.java]: @yegor256 You're right. I've been changed them. Thanks!

File: src/main/java/org/cactoos/text/SubText.java (hunk 1 above)
Comment (fabriciofx) [inline src/main/java/org/cactoos/text/SubText.java]: @yegor256 I'm sorry, but I didn't get it. The problem is the `Exception`? I put it due the methods `this.start.asValue()` and `this.end.asValue()` that throw an `Exception`.

Comment (yegor256) [inline src/main/java/org/cactoos/text/SubText.java]: @fabriciofx you don't to `catch`, just wrap `Text` in `UncheckedText`

File: src/test/java/org/cactoos/text/SubTextTest.java (hunk 2 above)
Comment (yegor256) [inline src/test/java/org/cactoos/text/SubTextTest.java]: @fabriciofx don't see the changes yet

**Comment by yegor256** [comment]
@fabriciofx see above. make sure you message me after you make changes. without that I miss your updates and the PR stays open for longer. read this: http://www.yegor256.com/2014/11/24/principles-of-bug-tracking.html

File: src/main/java/org/cactoos/text/SubText.java (hunk 1 above)
Comment (fabriciofx) [inline src/main/java/org/cactoos/text/SubText.java]: @yegor256 Wrap `Text` in `UncheckedText` or `start` and `end` in `UncheckedScalar`? I did the last one, and removed the catch. So, could you check if it's ok now?

**Comment by fabriciofx** [comment]
@yegor256 You're completely right. I'll review all my tickets/PR to close them faster as I can. Thanks for your attention and patience!

File: src/test/java/org/cactoos/text/SubTextTest.java (hunk 2 above)
Comment (fabriciofx) [inline src/test/java/org/cactoos/text/SubTextTest.java]: @yegor256 I'm sorry. I fix it now.

**Comment by fabriciofx** [comment]
@yegor256 Could you review it now? Thanks in advance!

**Comment by yegor256** [comment]
@rultor merge

**Pull request merged**